qt/RpcQueue.h

```diff
@@ -0,0 +1,90 @@
+/*
+ * This file Copyright (C) 2015 Mnemosyne LLC
+ *
+ * It may be used under the GNU GPL versions 2 or 3
+ * or any future license endorsed by Mnemosyne LLC.
+ *
+ * $Id$
+ */
+
+#ifndef QTR_RPC_QUEUE_H
+#define QTR_RPC_QUEUE_H
+
+#include <assert.h>
+
+#include <QQueue>
+#include <QFutureWatcher>
+
+#include "RpcClient.h"
+
+class RpcQueue: public QObject
+{
+  Q_OBJECT
+
+  /*
+   * Internally queued function. Takes the last response future, makes a
+   * request and returns a new response future.
+   */
+  typedef std::function<QFuture<RpcResponse> (QFuture<RpcResponse>)> InternalQueuedFunction;
```

Comment:
Not exactly, should be included here in .h, where std::function<> is being used, not in .cc.